packages/grid/_modules_/grid/models/gridOptions.tsx

```diff
@@ -397,6 +397,10 @@ export interface GridOptions {
    * Set it to 'server' if you would like to handle the pagination on the server-side.
    */
   paginationMode?: GridFeatureMode;
+  /**
+   * Function that allows to apply a class name on each row.
+   */
+  rowClassName?: (params: GridRowParams) => string;
```

Comment:
if we only allow a function then the name is misleading, it should be `getRowClassName`
I think it could also handle `string`and `string[]`

Reply:
cellClassName in ColDef is of type `GridCellClassNamePropType`
these 2 prop should work the same way

Reply:
Should we move cellClassName  from colDef to gridOptions?
Also we can remove cellClassRules

Reply:
👍  for `getRowClassName`, a static classes comes with the `classes` API. AG Grid has two props: https://ag-grid.com/react-grid/row-styles/#row-style as well

Reply:
how do you add a class on rows with `classes`?

Reply:
Developers would do:

```jsx
<DataGrid classes={{ row: 'my-class-name' }} /> 
```

Reply:
I renamed to `getRowClassName`. For static class names, the `classes` prop can be used. We'll have to rename `cellClassName` and `headerClassName` in another PR to keep consistency.